### DISCUSSION TITLE
Option to print diagnostics more compactly

### DISCUSSION THREAD

**Discussion by andenacitelli** [biomejs/biome#1909] [Ideas]
Logs seem to have `--log-kind` which can be set to `"pretty" | "compact" | "json"`. However, as referenced in [a related issue](https://github.com/biomejs/biome/issues/1701), a similar option is not available for the printing of actual diagnostics.

When I initially tried biome, this threw me off. I ran on an existing project I am considering migrating from ESLint (meaning lots of initial errors popped up -- though to Biome's credit, many were autofixable) and it was just a _lot_ of output.

Each error simply takes up a _lot_ of vertical lines. Here's an example:

```txt
./src/pages/index.tsx:66:2 lint/correctness/useExhaustiveDependencies ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ This hook does not specify all of its dependencies: userFilters.isSuccess
  
    64 │        }, [userFilters.data]);
    65 │ 
  > 66 │        useEffect(() => {
       │        ^^^^^^^^^
    67 │                //* Process form value changes
    68 │                console.log(form.values);
  
  ℹ This dependency is not specified in the hook dependency list.
  
    76 │                setFilters(form.values);
    77 │                if (
  > 78 │                        userFilters.isSuccess &&
       │                        ^^^^^^^^^^^^^^^^^^^^^
    79 │                        JSON.stringify(userFilters.data) !== JSON.stringify(form.values)
    80 │                ) {

```

This is **19 lines of output** for one error.

While I think the verbose descriptions are a benefit, this was simply a ton of output and took a while to scroll up, running it fresh on an existing project I'm migrating from ESLint. Perhaps allowing parameterization of one or more things might be useful:

- Number of lines of code printed (or whether that code is printed at all)
- Whether the lines of pure whitespace (for example, the lines immediately before and after `✖ This hook does not specify all of its dependencies: userFilters.isSuccess`) should be included

I'd think a similar interface where users can supply `"pretty" | "compact"` would be the best user experience. Pretty would be the existing option, compact would essentially get rid of the pure whitespace lines and trim (or entirely remove) the code print-out.

**Comment by Conaclos** [comment]
This could be partially solved by a `reporter` flag as discussed in #57.
